include/common/nas_structs.h

```diff
@@ -411,6 +411,86 @@ typedef struct tai_list {
 	struct TAI partial_list[EMM_MAX_TAI_LIST];
 } tai_list;
 
+/* 24.301 - 9.9.4.3.1 */
+typedef struct eps_qos_bitmap {
+	uint8_t qci;
+        uint8_t mbr_ul;
+        uint8_t mbr_dl;
+        uint8_t gbr_ul;
+        uint8_t gbr_dl;
+        uint8_t mbr_ul_ext;
+        uint8_t mbr_dl_ext;
+        uint8_t gbr_ul_ext;
+        uint8_t gbr_dl_ext;
+        uint8_t mbr_ul_ext_2;
+        uint8_t mbr_dl_ext_2;
+        uint8_t gbr_ul_ext_2;
+        uint8_t gbr_dl_ext_2;
+} eps_qos_bitmap;
+
+typedef struct eps_qos_t {
+        uint8_t len;
+	eps_qos_bitmap val;
+} eps_qos_t;
+
+#define TFT_MAX_SIZE 256
+
+typedef struct bearer_tft{
+    uint16_t len;
+    uint8_t data[TFT_MAX_SIZE];
+} bearer_tft;
+
+typedef enum esm_cause_t {
```

Comment:
Would be good to have spec and section reference. That becomes handy for new folks

Reply:
Taken care.